client/src/components/Calendar/StaticCalendar.js

```diff
@@ -46,30 +55,91 @@ var weekString =
 export default class StaticCalendar extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { shifts: initialShifts() };
+    this.state = { shifts: initialShifts(), modalIsOpen: false };
+    this.stateFixer = this.stateFixer.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+    this.openModal = this.openModal.bind(this);
+    this.submitClick = this.submitClick.bind(this);
+    this.cancelClick = this.cancelClick.bind(this);
+  }
+
+  closeModal() {
+    this.setState({ modalIsOpen: false });
+  }
+
+  openModal() {
+    this.setState({ modalIsOpen: true });
   }
 
   componentDidMount() {
     fetch("/staticcalendar/" + this.props.userId, {
       method: "POST",
       headers: {
         Accept: "application/json",
-        "Content-Type": "application/json"
+        "Content-Type": "application/json",
       },
       body: JSON.stringify({
-        items: this.state.shifts, userId: this.props.userId
-      })
+        items: this.state.shifts,
+        userId: this.props.userId,
+      }),
     })
-      .then(response => {
+      .then((response) => {
         console.log("response");
         return response.json();
       })
-      .then(jsonResponse => {
+      .then((jsonResponse) => {
         console.log(jsonResponse.shifts);
         this.setState({ shifts: jsonResponse.shifts });
       });
   }
 
+  stateFixer(e) {
+    if (
+      e.target.id != "" &&
+      e.target.style.backgroundColor != "rgb(193, 135, 211)"
+    ) {
+      /*The point of the background color check is to make sure that once a shift is requested to be covered,
+       * this can't happen again for the same shift.
+       */
+      console.log(e.target.style.backgroundColor);
+      currentClicked = e;
+      currentClickedID = e.target.id;
+      this.openModal();
+    }
+  }
+  submitClick() {
+    var reason = document.getElementById("reason");
+    var notes = reason.value;
+    let newShifts = this.state.shifts;
+    for (let i = 0; i < newShifts.length; i++) {
+      if (newShifts[i].id == currentClickedID) {
```

Comment:
Lets only do this stuff after the server responds, because if the server crashes before the request makes it there, we don't want to tell the user their coverrequest went through. Just copy and paste this into the .then(jsonresponse) function

Reply:
resolved